feat: add basic logic to despawn enemies when hit by a projectile

campenr opened the pull request:
Adds:
- on hit enemies and the projectile despawn
- enemies despawn when off the LHS of the screen + buffer

Checkroth commented on units/enemy.gd:
```diff
@@ -2,7 +2,12 @@ extends Node2D
 	position -= transform.x * SPEED * delta
+	
+	# despawn enemies that have moved beyond the left of the screen
+	if global_position.x < DESPAWN_BOUND:
```
https://docs.godotengine.org/en/stable/classes/class_visibleonscreennotifier2d.html

we should avoid re-inventing the wheel :grin:

campenr replied:
If nothing else this game dev is super helpful to learn how to actually use godot :nods:

campenr committed:
feat: add basic logic to remove enemies when hit or offscreen

- projectiles hitting enemies cause both to be freed
- enemies offscreen are freed (using notifier node!! thanks @chekroth)

campenr commented on project.godot:
```diff
@@ -15,6 +15,10 @@ run/main_scene="res://levels/game.tscn"
 config/features=PackedStringArray("4.3", "GL Compatibility")
 config/icon="res://icon.svg"
 
+[display]
```
🍷 🧀 Helps with debugging

campenr commented on units/enemy.tscn:
```diff
@@ -4,7 +4,7 @@
 
 [sub_resource type="RectangleShape2D" id="RectangleShape2D_mvxux"]
-size = Vector2(53, 26)
+size = Vector2(86, 45.5)
```
🍷 🧀 I guess I moved it? 🤷

Checkroth commented on units/projectile.gd:
```diff
@@ -2,5 +2,12 @@ extends Area2D
+func _on_area_entered(area):
+	# TODO: apply effects on enemy hit. For now just despawn.
+	queue_free()
+	area.owner.queue_free()
```
what is the `area.owner` actually, here?

campenr replied:
An `Enemy` node, so that we remove the whole enemy and not just the `Area2D` collision box.
But yes, this feels a little 🪄 doesn't it. A next step would be to add a method to the collider for handling what to do, and we could just call that as `area.handle_projectile_collision` or something.

Checkroth replied:
Doesn't `queue_free` clear the whole scene? (at least below the node?)

So instead of `owner.queue_free`, we should be emitting a "hit" event here or something, and the `Enemy` node (or whatever is the root of the scene) should handle the queue_free-ing

But that's just details. :ship: :rocket: `merge-when-approved`

Checkroth reviewed (approved):
I think we can improve how we use the godot events system, but if it fits I sits
